### PR TITLE
Users can change pull request url

### PR DESCRIPTION
**Pull request by ukpagrace**
Modified Code to Allow users to change their pull request
Screenshot (50)
Screenshot (51)
Screenshot (49)

### CODE REVIEW

File: TCSA.V2/Components/Dashboard/Pages/ProjectSubmission.razor [hunk 1]
```diff
@@ -13,13 +13,57 @@
...
+                    <button class="btn btn-outline-secondary mt-1" type="submit">Update GitHubUrl</button>
+                }
+
+            </EditForm>
         </div>
-    </div>
+    }
+    else if(IsSubmissionSuccessful)
+    {
+        <div class="mt-5 pt-5">
```
Comment (dmarcoux) [inline TCSA.V2/Components/Dashboard/Pages/ProjectSubmission.razor:48]: With the introduction of these changes, I would suggest removing some of the duplication in the HTML markup in this page. A sizable part of the HTML is always the same in every `if/else` branch:

```html
        <div class="mt-5 pt-5">
            <div class="card submit-project-card bg-@(backgroundClass)">
                <div class="card-body">
                    @cardContent
                </div>
            </div>
        </div>
``` 

Above is only an example of what could be abstracted. The background class and maybe even the card content, which ends up being a short message in most `if/else` branches.

File: TCSA.V2/Components/Dashboard/Pages/ProjectSubmission.razor [hunk 2]
```diff
@@ -13,13 +13,57 @@
...
+                </div>
+            </div>
+        </div>
+    }
+    else
+    {
+        <div class="mt-5 pt-5">
+            <div class="card submit-project-card bg-danger">
+                <div class="card-body">
+                    This Project has been Marked Completed
```
Comment (dmarcoux) [inline TCSA.V2/Components/Dashboard/Pages/ProjectSubmission.razor:61]: To use a consistent case:

```suggestion
                    This project has been marked as completed.
```

File: TCSA.V2/Components/Dashboard/Pages/ProjectSubmission.razor (hunk 1 above)
Comment (ukpagrace) [reply TCSA.V2/Components/Dashboard/Pages/ProjectSubmission.razor:48]: I'm thinking, Abstracting card content would be putting it in a component file. Isn't it too small for it?

Comment (dmarcoux) [reply TCSA.V2/Components/Dashboard/Pages/ProjectSubmission.razor:48]: We could always wait to see how it goes. If we end up having another instance of this HTML in the future, then I would suggest going with a component. Even if it's quite small.

File: TCSA.V2/Components/Dashboard/Pages/ProjectSubmission.razor (hunk 2 above)
Comment (ukpagrace) [reply TCSA.V2/Components/Dashboard/Pages/ProjectSubmission.razor:61]: noted, thanks.

File: TCSA.V2/Components/Dashboard/Pages/ProjectSubmission.razor (hunk 1 above)
Comment (ukpagrace) [reply TCSA.V2/Components/Dashboard/Pages/ProjectSubmission.razor:48]: okay

Comment (TheCSharpAcademy) [reply TCSA.V2/Components/Dashboard/Pages/ProjectSubmission.razor:48]: @dmarcoux Thanks for your input! It's always good to have another pair of experienced eyes in these reviews! 👌🏻👌🏻😁

**Review by TheCSharpAcademy — CHANGES_REQUESTED** [review]
@ukpagrace thanks for submitting! At the moment I can update the PR but I'll request two changes for better UX.

1 - In the dashboard main page: change the button label to view/edit instead of just view, otherwise it's not intuitive that the PR can be updated from there. Also, it would be good if the icon itself was clickable too. And I'm the card for the project underreview should have a yellow border. 

2 - This is optional, if you think it's too much work I can create a separate ticket. But you could create a button similar to Show Completed, but Show Pending.

If that button is clicked you need change the button show it reverses the action. Also, you'd need to show a list of Project Cards, as opposed to Area Cards. Let me know if you want to work on this of if I should create a separate ticket. 😄

Is that ok? Thank you!

**Comment by ukpagrace** [comment]
@TheCSharpAcademy Thanks for the detailed review.
Noted on the required changes and yes I would on it.

**Comment by stevietv** [comment]
> @ukpagrace thanks for submitting! At the moment I can update the PR but I'll request two changes for better UX.
> 
> 1 - In the dashboard main page: change the button label to view/edit instead of just view, otherwise it's not intuitive that the PR can be updated from there. Also, it would be good if the icon itself was clickable too. And I'm the card for the project underreview should have a yellow border.

@TheCSharpAcademy 
I think it should be the pending button that becomes `Pending/Edit` not the view button?
the NavLink should indeed include the SVG for the icon inside it.

**Pull request merged**